aptos-move/framework/aptos-stdlib/sources/table.move

```diff
@@ -93,6 +71,11 @@ module aptos_std::table {
         drop_unchecked_box<K, V, Box<V>>(table)
     }
 
+    public(friend) fun destroy<K: copy + drop, V>(table: Table<K, V>) {
```

Comment:
raw table can't be deleted now?

Reply:
that would be unsafe... maybe @wrwg can comment on that though

Reply:
oh because of dangling items

Reply:
Right, the API kinda makes it seem like you can delete it, but you take the consequences of doing so... given that we do secondary indexing. I'm inclined to not support deletes. We can always bring support of the secondary index to move and check to see if it is empty, but that's not going to happen now and it is a non-breaking API change :)

Reply:
I'm not sure what you are talking about. What is raw table? And what would be unsafe here?

Reply:
raw table -> deleting the handle and the underlying resources, would this have any impact on say storage or indexers if we just eliminate the table but their are dangling references. @msmouse 

We  can always expose this later.

Reply:
On Move side their can be no dangling references because this thing is linear, but not sure about storage.

As regards destroy should this API now succeed even if the table isn't empty? I would consider this as problematic.

Reply:
That's why I want to have the TableWithLength and friends. We can program it so that it we know it won't destroy unless the table is indeed empty.

Reply:
Sorry I'm late to the party. I think this implementation is proper: the underlying table is delete-able only by the TableWithLength who knows its empty.